mantidimaging/gui/windows/spectrum_viewer/spectrum_widget.py

```diff
@@ -181,23 +197,15 @@ def set_roi_visibility_flags(self, name: str, visible: bool) -> None:
         @param name: The name of the ROI.
         @param visible: The new visibility of the ROI.
         """
-        handles = self.roi_dict[name].getHandles()
-        for handle in handles:
-            handle.setVisible(visible)
-        self.roi_dict[name].setVisible(visible)
+        self.roi_dict[name].set_visibility(visible)
 
     def set_roi_alpha(self, name: str, alpha: float) -> None:
         """
-        Change the alpha value of an existing ROI
-
-        @param name: The name of the ROI.
-        @param alpha: The new alpha value of the ROI.
+        Change the alpha value of an existing ROI and update visibility.
         """
-
-        self.roi_dict[name].colour = self.roi_dict[name].colour[:3] + (alpha, )
-        self.roi_dict[name].setPen(self.roi_dict[name].colour)
-        self.roi_dict[name].hoverPen = mkPen(self.roi_dict[name].colour, width=3)
-        self.set_roi_visibility_flags(name, bool(alpha))
```

Comment:
Given that this is about to go away I would not put the effort in to moving it to the `SpectrumROI`.

I think it could simply to
```
self.roi_dict[name].colour = self.roi_dict[name].colour[:3] + (alpha, )
self.roi_dict[name].set_visibility(bool(alpha))
```
and maybe just
```
self.roi_dict[name].set_visibility(bool(alpha))
```
because if we have set the visibility to false, it does not matter what the alpha is.